### PR TITLE
Make first reservable time consider reservations

### PR DESCRIPTION
**Pull request by matti-lamppu**
## 🛠️ Changelog
[//]: # "Describe the changes in this pull request here."

- Makes the first reservable time consider reservations from reservation units with a common resource or space "through hierarchy"

## 🧪 Test plan
[//]: # "Help your fellow reviewer and write a short description of what's the fastest way to test your changes."

- Automated tests
- Manual tests when in the test server

## 🎫 Tickets
[//]: # "This pull request resolves all or part of the following ticket(s)"

- [TILA-2821](https://helsinkisolutionoffice.atlassian.net/browse/TILA-2821), [TILA-2852](https://helsinkisolutionoffice.atlassian.net/browse/TILA-2852)

[TILA-2821]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-2821?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TILA-2852]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-2852?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by matti-lamppu** [comment]
@vergama Went through the changes and made my corrections, so this is OK by me. @ranta had already finished everything by the end of the year, so not much changed. Tests were also in place and should be enough for now at least.

File: tests/test_querysets/test_reservation_querysets.py [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+        )
+    }
+
+    assert all_closed_time_spans.get(reservation_unit_id) == closed_time_spans
+
+
+def test__get_affecting_reservations__only_resources():
+    # Resource 1 <- ReservationUnit 1, 2
+    # Resource 2 <- ReservationUnit 3, 4
+    # Resource 2 <- ReservationUnit 5
```
Comment (vergama) [inline tests/test_querysets/test_reservation_querysets.py:78]: Typo, should read: # Resource **3** <- ReservationUnit 5

File: spaces/querysets/space.py [hunk 2]
```diff
@@ -1,63 +1,35 @@
...
 ]
 
 
-class SpaceQuerySet(TreeQuerySet):
-    def all_space_ids_though_hierarchy(self) -> set[int]:
+class SpaceQuerySet(ExtendedTreeQuerySet):
+    def all_space_ids_though_hierarchy(self: Self) -> set[int]:
         """
-        Get ids for all spaces that are accessible though the space hierarchy
+        Get ids of all spaces that are accessible though the space hierarchy
```
Comment (vergama) [inline spaces/querysets/space.py:13]: Old typo here: though >> th**r**ough

File: tests/test_querysets/test_reservation_querysets.py (hunk 1 above)
Comment (matti-lamppu) [reply tests/test_querysets/test_reservation_querysets.py]: fixed

File: spaces/querysets/space.py (hunk 2 above)
Comment (matti-lamppu) [reply spaces/querysets/space.py]: fixed

**Review by vergama — APPROVED** [review]
Looks good!

**Pull request merged**